nb/projects/belt_sprockets/belt_sprockets.nb

```diff
@@ -0,0 +1,741 @@
+      [0.90364, 0.217664, 0],
+      [0.932921, 0.130969, 0],
+      [0.988924, 0.062001, 0],
+      [1.065168, 0.016448, 0],
+      [1.155171, 0, 0],
+      [1.155171, -0.5, 0],
+    ]
+  ),
+  diameter: (teeth) => ToothSpacing(teeth, 3, 0.381),
+  toothDepth: 1.169,
+  toothWidth: 2.31,
+};
+
+const GT2_5mm = {
+  profile: ToothProfileToFace(
+    ...[
+      [-1.975908, -0.75, 0],
+      [-1.975908, 0, 0],
+      [-1.797959, 0.03212, 0],
+      [-1.646634, 0.121224, 0],
+      [-1.534534, 0.256431, 0],
+      [-1.474258, 0.426861, 0],
+      [-1.446911, 0.570808, 0],
+      [-1.411774, 0.712722, 0],
+      [-1.368964, 0.852287, 0],
+      [-1.318597, 0.989189, 0],
+      [-1.260788, 1.123115, 0],
+      [-1.195654, 1.25375, 0],
+      [-1.12331, 1.380781, 0],
+      [-1.043869, 1.503892, 0],
+      [-0.935264, 1.612278, 0],
+      [-0.817959, 1.706414, 0],
+      [-0.693181, 1.786237, 0],
+      [-0.562151, 1.851687, 0],
+      [-0.426095, 1.9027, 0],
+      [-0.286235, 1.939214, 0],
+      [-0.143795, 1.961168, 0],
+      [0, 1.9685, 0],
+      [0.143796, 1.961168, 0],
+      [0.286235, 1.939214, 0],
+      [0.426095, 1.9027, 0],
+      [0.562151, 1.851687, 0],
+      [0.693181, 1.786237, 0],
+      [0.817959, 1.706414, 0],
+      [0.935263, 1.612278, 0],
+      [1.043869, 1.503892, 0],
+      [1.123207, 1.380781, 0],
+      [1.195509, 1.25375, 0],
+      [1.26065, 1.123115, 0],
+      [1.318507, 0.989189, 0],
+      [1.368956, 0.852287, 0],
+      [1.411872, 0.712722, 0],
+      [1.447132, 0.570808, 0],
+      [1.474611, 0.426861, 0],
+      [1.534583, 0.256431, 0],
+      [1.646678, 0.121223, 0],
+      [1.798064, 0.03212, 0],
+      [1.975908, 0, 0],
+      [1.975908, -0.75, 0],
+    ]
+  ),
+  diameter: (teeth) => ToothSpacing(teeth, 5, 0.5715),
+  toothDepth: 1.969,
+  toothWidth: 3.952,
+};
+
+const Pulley = (profile, teeth) => {
+  const pulley_OD = profile.diameter(teeth);
+  const tooth_depth = profile.toothDepth;
+  const tooth_width = profile.toothWidth;
+  const toothDistanceFromCentre = Math.sqrt(
+    Math.pow(pulley_OD / 2, 2) -
+      Math.pow((tooth_width + additionalToothWidth) / 2, 2)
+  );
+  const toothWidthScale = (tooth_width + additionalToothWidth) / tooth_width;
+  const toothDepthScale = (tooth_depth + additionalToothDepth) / tooth_depth;
+  var toothProfile = profile.profile;
+
+  console.log('pulley_OD: %f', pulley_OD);
+  console.log(profile);
+
+  const center = Arc(pulley_OD, pulley_OD);
+
+  //teeth - cut out of shaft
+
+  //Scale tooth up
+  toothProfile = toothProfile.scale(toothWidthScale, toothDepthScale);
+  //Translate it out to where it should be
+  toothProfile = toothProfile.move(0, -toothDistanceFromCentre, 0);
+
+  var teethArray = [];
+  for (let i = 0; i < teeth; i++) {
+    teethArray.push(toothProfile.rz(i / teeth));
+  }
+  const teethToCut = Group(...teethArray);
+
+  return center.cut(teethToCut);
+};
+
+md`
+# Supported belt profiles:`;
+
+
+
+
+
+
+md`
+MXL`;
+Pulley(MXL, 20).view();
```

Comment:
If we're computing these, we might as well export definitions.

How about

export const xml20Pulley = Pulley(XML, 20).view().md('MXL'); ?

Or md('XML').view(); if you want the label above.

Reply:
Are they really useful? What are the odds that someone needs exactly 20 teeth?

Reply:
The change in how the labels are generated is nice and much cleaner

Reply:
Hmm, good point. :)